### PR TITLE
Add SE3LayerNorm, SE3RadialFunc and SE3PairwiseConv layers for SE3-Transformer model

### PR DESCRIPTION
**Pull request by JoseAntonioSiguenza**
## Description

Initial implementation of SE3PairwiseConv, SE3LayerNorm, and SE3RadialFunc layers for the SE(3)-Transformer model. PairwiseConv ensures SE(3)-equivariant feature interactions, decomposing them into radial (distance-based) and angular (spherical harmonics) components. SE3LayerNorm applies per-channel normalization for stability, while SE3RadialFunc learns flexible distance-based interactions with batch normalization. This PR has been through an intial review at #4295 

## Type of change

Please check the option that is related to your PR.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
  - In this case, we recommend to discuss your modification on GitHub issues before creating the PR
- [ ] Documentations (modification for documents)

## Checklist

- [x] My code follows [the style guidelines of this project](https://deepchem.readthedocs.io/en/latest/development_guide/coding.html)
  - [x] Run `yapf -i <modified file>` and check no errors (**yapf version must be  0.32.0**)
  - [x] Run `mypy -p deepchem` and check no errors
  - [x] Run `flake8 <modified file> --count` and check no errors
  - [x] Run `python -m doctest <modified file>` and check no errors
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New unit tests pass locally with my changes
- [x] I have checked my code and corrected any misspellings

### CODE REVIEW

File: deepchem/models/torch_models/tests/test_se3_transformer.py [hunk 1]
```diff
@@ -22,6 +51,100 @@ def rotation_matrix(axis, angle):
     ]])
 
 
+def apply_rotation(x, axis, angle):
```
Comment (rbharath) [inline deepchem/models/torch_models/tests/test_se3_transformer.py:54]: Please add type annotations.

File: deepchem/models/torch_models/tests/test_se3_transformer.py [hunk 2]
```diff
@@ -22,6 +51,100 @@ def rotation_matrix(axis, angle):
...
+    """Apply a 3D rotation to the positions."""
+    R = rotation_matrix(axis, angle)
+    return torch.tensor(np.dot(x.numpy(), R), dtype=torch.float32)
+
+
+def get_equivariant_basis(G, max_degree):
```
Comment (rbharath) [inline deepchem/models/torch_models/tests/test_se3_transformer.py:60]: Type annotation and docstring please

Comment (rbharath) [reply deepchem/models/torch_models/tests/test_se3_transformer.py:60]: This feels like it should be a utility function

Comment (rbharath) [reply deepchem/models/torch_models/tests/test_se3_transformer.py:60]: This can be done in the next PR

**Review by rbharath — APPROVED** [review]
LGTM
